core/src/main/scala/org/apache/spark/BarrierCoordinator.scala

```diff
@@ -187,6 +191,12 @@ private[spark] class BarrierCoordinator(
       requesters.clear()
       cancelTimerTask()
     }
+
+    // Check for clearing internal data, visible for test only.
+    private[spark] def cleanCheck(): Boolean = requesters.isEmpty && timerTask == null
+
+    // Get currently barrier epoch, visible for test only.
+    private[spark] def getBarrierEpoch(): Int = barrierEpoch
```

Comment:
Why not just make `barrierEpoch` visible for testing?

Reply:
https://github.com/apache/spark/pull/22165#discussion_r218093991 As the comment here, need revert back?